### PR TITLE
chore(STONEO11Y-237): Code-review guide

### PR DESCRIPTION
**Pull request by Omeramsc**
Adds a guide with basic principles and tips for code-review to assist our team and newcomers during the review process and help make it healthy and useful.

### CODE REVIEW

File: review.md
Comment (yftacherzog) [inline review.md:1]: I'd suggest breaking all lines after ~90 chars

File: review.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+2. A great and fast way for both the reviewer and the code owner to share ideas, learn, and improve their code writing and understanding skills.
+
+3. Allow us as reviewers to get to know parts of the code base we usually don't work on.
+
+4. Help our fellow team members to validate their code quality and push updates faster.
+
+## Tips for reviewing
+
+-   It is important to maintain a healthy, respectful, and civil environment. Feel free to ask questions, offer suggestions, or provide criticism, regardless of your position or level of expertise. However, it is crucial to avoid personal attacks and focus on reviewing the code rather than criticizing the individual who wrote it. Additionally, please refrain from using capital letters, using inappropriate language, or addressing others directly.\
```
Comment (yftacherzog) [inline review.md:15]:
```suggestion
-   It is important to maintain a healthy, respectful, and civil environment. Feel free to ask questions, offer suggestions, or provide criticism, regardless of your position or level of expertise. However, it is crucial to avoid personal attacks and focus on reviewing the code rather than criticizing the individual who wrote it. Additionally, please refrain from using capital letters (all caps), using inappropriate language, or addressing others directly.

```

Comment (yftacherzog) [reply review.md:15]: What does `addressing others directly` mean?

File: review.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+| "This won't work, maybe try and look into X"                                                                       | "THIS IS WRONG."                             |
+| "Missing a test for X",                                                                                            | "What is it with you and missing tests?"     |
+| "I've seen this error repeated multiple times, have I mentioned X? try to look into it, it has some nice tools"    | "You keep repeating those types of mistakes" |
+| What is the reason you chose this implementation?                                                                  | huh? Why did you do that?                    |
+
+
+-   Don't forget that in the worst case scenario - you're wrong, and you learned something for the next time.
+
+
+- Remember that the goal isn't solely to suggest changes; it can also serve as an opportunity to ask questions. By asking questions, you can challenge the code owner and assist them in gaining a deeper understanding of their own code. When providing answers, code owners often realize the need for changes they hadn't considered before. Engaging in a discussion will benefit both you and the code owner.
```
Comment (yftacherzog) [inline review.md:32]:
```suggestion
- Remember that the goal isn't solely to suggest changes; it can also serves as an opportunity to ask questions. By asking questions, you can challenge the code owner and assist them in gaining a deeper understanding of their own code. When providing answers, code owners often realize the need for changes they hadn't considered before. Engaging in a discussion will benefit both you and the code owner.   
```

File: review.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    Here are some aspects to consider during the review process:  
+    - Pay attention to edge cases, such as race conditions, cleanups, and the presence of tests that cover enough use cases without becoming excessive. 
+    - Be on the lookout for missing tests. A good rule of thumb is that if something requires manual execution to see that it works, it's possible that corresponding tests are absent. 
+    - Assess whether the PR focuses on a specific subject. Lengthy or disorganized PRs can make the review process more challenging. It may be beneficial to break them down into smaller, standalone PRs, making it easier to identify errors and shorten the feedback loop. 
+    - Verify that the added functionality does not already exist within the codebase. Take into consideration existing functions, features, or packages that could be utilized instead of reinventing the wheel.
```
Comment (yftacherzog) [inline review.md:49]: You have `code base` and `codebase` on this file. I'd suggest sticking with one of them.

File: review.md (hunk 1 above)
Comment (Omeramsc) [reply review.md:15]: Using "You", for instance. The examples put more focus on it.

File: review.md (hunk 2 above)
Comment (Omeramsc) [reply review.md:32]: I'm pretty sure the way I wrote it is correct, unless we remove the "can", in that case "serves" will fit

File: review.md (hunk 3 above)
Comment (Omeramsc) [reply review.md]: done

File: review.md (hunk 1 above)
Comment (Omeramsc) [reply review.md]: done

File: review.md (hunk 2 above)
Comment (Omeramsc) [reply review.md]: anyway, done

Comment (yftacherzog) [reply review.md]: You're obviously right. not sure what I had in mind.

**Comment by ShakedAviv50313** [comment]
LGTM

File: review.md (hunk 1 above)
Comment (yftacherzog) [reply review.md]: > Using "You", for instance. The examples put more focus on it.

I think I see what you mean. We should talk about the code rather than who wrote it, but I think this expectation is a bit extreme.

File: review.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
+# Code Review principles and tips
+
+## Why do we code review?
+
+1. Code reviews are essential to improve the level of our code, and to prevent bugs. 
+
+2. A great and fast way for both the reviewer and the code owner to share ideas,
```
Comment (yftacherzog) [inline review.md:7]: extra space here at the end is breaking the line on the rendered text as well
```suggestion
2. A great and fast way for both the reviewer and the code owner to share ideas, 
```

File: review.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
+# Code Review principles and tips
+
+## Why do we code review?
+
+1. Code reviews are essential to improve the level of our code, and to prevent bugs.
```
Comment (yftacherzog) [inline review.md:5]:
```suggestion
1. Code reviews are essential to improve the level of our code, and to prevent bugs.
```

File: review.md [hunk 6]
```diff
@@ -0,0 +1,92 @@
+# Code Review principles and tips
+
+## Why do we code review?
+
+1. Code reviews are essential to improve the level of our code, and to prevent bugs. 
+
+2. A great and fast way for both the reviewer and the code owner to share ideas,  
+   learn, and improve their code writing and understanding skills.
+
+3. Allow us as reviewers to get to know parts of the code base we usually
```
Comment (yftacherzog) [inline review.md:10]:
```suggestion
3. Allow us as reviewers to get to know parts of the code base we usually
```

File: review.md [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+
+## Tips for reviewing
+
+-   It is important to maintain a healthy, respectful, and civil environment. Feel free  
+    to ask questions, offer suggestions, or provide criticism, regardless of your  
+    position or level of expertise. However, it is crucial to avoid personal attacks  
+    and focus on reviewing the code rather than criticizing the individual who wrote it.  
+    Additionally, please refrain from using capital letters (all caps), using  
+    inappropriate language, or addressing others directly.  
+    Here are a few examples:
```
Comment (yftacherzog) [inline review.md:23]: I suggest remove the spaces at end of line. The idea being that we control the line length for the code, which we let the rendered text to be as long as the screen it is read from allows.
```suggestion
-   It is important to maintain a healthy, respectful, and civil environment. Feel free
    to ask questions, offer suggestions, or provide criticism, regardless of your
    position or level of expertise. However, it is crucial to avoid personal attacks
    and focus on reviewing the code rather than criticizing the individual who wrote it.
    Additionally, please refrain from using capital letters (all caps), using
    inappropriate language, or addressing others directly.
    Here are a few examples:
```

File: review.md
Comment (yftacherzog) [reply review.md:1]: The idea behind this comment was to limit the line length within the text file, while letting the rendering environment control the length of the rendered output.

To do that, we leave no spaces at and of lines, and just break them with a single newline.

This way the text file will fit inside a standard editor screen while the rendered output will fit the window in which it's is rendered.

Comment (Omeramsc) [reply review.md:1]: Thanks, misunderstood the intention originally.

File: review.md (hunk 1 above)
Comment (Omeramsc) [reply review.md]: I thought the examples would make the expectation clearer... how would you rephrase it? I still think creating a connection between the way the code is written and who wrote it, in a direct way, as shown in the examples, is something we should be thoughtful of.

File: review.md (hunk 7 above)
Comment (Omeramsc) [reply review.md]: fixed

File: review.md (hunk 6 above)
Comment (Omeramsc) [reply review.md]: fixed

File: review.md (hunk 4 above)
Comment (Omeramsc) [reply review.md]: fixed

File: review.md (hunk 1 above)
Comment (yftacherzog) [reply review.md]: I think we already mention we should focus on the code rather than the individual. I think that if you drop this part here and leave the examples as-is, it should be enough.

**Review by yftacherzog — APPROVED** [review]

**Review by ShakedAviv50313 — APPROVED** [review]
